### DISCUSSION TITLE
Caching with keys

### DISCUSSION THREAD

**Discussion by wm-simon** [statamic/cms#6795] [Q&A]
Hey,

I have a site, where the users can filter for products.
The website is cached `half`. For the filter section I use the `{{ nocache }}` tag, otherwise it would not work.
The filter works with url params, like this:
`https://kabelmat.com/produktfinder?funktion=lagern|transportieren&verschiedenes=konfigurierbar&betriebsart=manuell`

The loading times are really high, so I've tried to cache the content by generated keys:
`{{ cache for="{ get | option_list | slugify }" }}`

But this seems to have no effect.

To generate the filter links dynamic I use this snippet, what is called about 10 times:
```
{{ filter_url = get }}

{{ foreach:get }}
{{ filter_url[key] = (get[$key] | explode:|) }}
{{ /foreach:get }}

{{ if filter_url[current_cat] }}

    {{ if filter_url[$current_cat] | contains($current_param) }}

    {{? 
    if (($delKey = array_search($current_param, $filter_url[$current_cat])) !== false) {
        unset($filter_url[$current_cat][$delKey]);
    } 
    ?}}

    {{ else }}

    {{? 
    array_push($filter_url[$current_cat], $current_param);
    ?}}

    {{ /if }}

    {{ if $filter_url[$current_cat] | count < 1 }}
    {{? 
        unset($filter_url[$current_cat]);
    ?}}
    {{ /if }}

{{ else }}

    {{? 
    $filter_url[$current_cat] = [$current_param];
    ?}}

{{ /if }}

?
{{ foreach:filter_url }}
{{ if filter_url[key] }}
{{ key }}=
{{ foreach:value  }}
{{ value }}{{ ! last ?= '|' }}
{{ /foreach:value }}
{{ ! last ?= '&' }}
{{ /if }}
{{ /foreach:filter_url }}
```

The collection contains about 100 entries with a depth of 3.

Does someone have an idea to get the loading times faster.

**Comment by edalzell** [comment]
I presume you're using a cache driver that supports keys?
